src/transaction/txParser.ts

```diff
@@ -43,13 +47,51 @@ const parseTxInputs = (
   return txInputs.map(([txHash, outputIndex]): _Input => ({ txHash, outputIndex }))
 }
 
-const parseTxOutputs = (
-  txOutputs: any[],
-): _Output[] => {
+const parseAssets = (assets: any): _Asset[] => {
+  if (!(assets instanceof Map)) {
+    throw Error(Errors.TxAssetParseError)
+  }
+  return Array.from(assets).map(([assetName, coins]) => {
+    if (!Buffer.isBuffer(assetName)) {
+      throw Error()
+    }
+    // TODO: is lovelace is not the best name since its not a lovelace
+    if (!isLovelace(coins)) {
+      throw Error()
+    }
+    return { assetName, coins: BigInt(coins) }
+  })
+}
+
+const parseMultiAsset = (multiAsset: any): _MultiAsset[] => {
+  if (!(multiAsset instanceof Map)) {
+    throw Error(Errors.TxMultiAssetParseError)
+  }
+  return Array.from(multiAsset).map(([policyId, assets]) => {
+    if (!Buffer.isBuffer(policyId)) {
```

Comment:
Don't we want to check the size of the buffer in cases where it is fixed? Ledger does that, and it is better  to report a meaningful error here vs. just "invalid data" thrown by Ledger.

Reply:
Good point, I think we want to check for the size of the buffers, certainly, it's nicer behavior than letting the device throw a general error. Initially, the assumption was that all txs come from the `cardano-cli` so naturally, they should be valid most of the time. If you don't see this as a crucial feature, I would make it into an issue and address it in another PR. What do you think?

Reply:
I agree, we will resolve this in another PR.

Reply:
I am getting this error . I don't know why

cardano-cli transaction assemble \
> --tx-body-file tx-pool.raw \
> --witness-file node.witness \
> --witness-file stake.witness \
> --witness-file payment.witness \
Invalid argument ` '
Did you mean this?
-h
Usage: cardano-cli transaction assemble --tx-body-file FILE
(--witness-file FILE) --out-file FILE